spark/dl/src/main/scala/com/intel/analytics/bigdl/parameters/AllReduceParameter.scala

```diff
@@ -135,6 +131,15 @@ class AllReduceParameter[T: ClassTag](
       .getOrElse(throw new IllegalStateException("Please initialize AllReduceParameter first!"))
   }
 
+  def localPartitionRange: (Int, Int) = {
+    val taskSize = size / partitionNum
+    val extraSize = size % partitionNum
+    val partitionId = TaskContext.getPartitionId()
```

Comment:
Are you sure the `partitionId` is the same as the `partitionId` of this object ?  use the global one instead

Reply:
This method should only be used in Rdd.mapPartitions(...). The partitionId here should be the id from the `mapPartitions`

Reply:
@wzhongyuan Changed as required. Found that `partitionId` is already valid when calling localPartitionRange